src/Features/LanguageServer/ProtocolUnitTests/Completion/CompletionResolveTests.cs

```diff
@@ -347,6 +347,43 @@ void M()
             Assert.NotNull(results.Description);
         }
 
+        [Theory, CombinatorialData]
+        public async Task TestSemanticSnippetChangeAsync(bool mutatingLspWorkspace)
+        {
+            var markup =
+                """
+                using System;
+                public class Program
+                {
+                    svm{|caret:|}
+                }
+                """;
+
+            await using var testLspServer = await CreateTestLspServerAsync(markup, mutatingLspWorkspace, new LSP.VSInternalClientCapabilities { SupportsVisualStudioExtensions = true });
+            testLspServer.TestWorkspace.GlobalOptions.SetGlobalOption(CompletionOptionsStorage.SnippetsBehavior, LanguageNames.CSharp, SnippetsRule.AlwaysInclude);
+            testLspServer.TestWorkspace.GlobalOptions.SetGlobalOption(CompletionOptionsStorage.ShowNewSnippetExperienceUserOption, LanguageNames.CSharp, true);
+
+            var caret = testLspServer.GetLocations("caret").Single();
+            var clientCompletionItem = await GetCompletionItemToResolveAsync<LSP.VSInternalCompletionItem>(testLspServer, label: "svm").ConfigureAwait(false);
+
+            Assert.True(clientCompletionItem.VsResolveTextEditOnCommit);
+
+            var results = (LSP.VSInternalCompletionItem)await RunResolveCompletionItemAsync(
+                testLspServer, clientCompletionItem).ConfigureAwait(false);
+
+            Assert.NotNull(results.TextEdit);
+            Assert.Null(results.InsertText);
+            Assert.Equal("static void Main(string[] args)\r\n    {\r\n        \r\n    }", results.TextEdit.Value.First.NewText);
+
+            var range = new LSP.Range
+            {
+                Start = new LSP.Position { Line = 3, Character = 4 },
+                End = new LSP.Position { Line = 3, Character = 7 }
+            };
```

Comment:
consider using markup for this.  i hate seeing random numbers in tests like thsi because i have no sense of what is represents wrt the final outcome, and if that seems correct or not.